e2e/test-component/scenarios/embedding-sdk/interactive-question.cy.spec.tsx

```diff
@@ -295,6 +295,44 @@ describeEE("scenarios > embedding-sdk > interactive-question", () => {
     });
   });
 
+  // NOTE: we will add more checks for other components to this test
+  // once we fix lifecycle warnings for them to prevent regressions.
+  it("should not show unsafe lifecycle warnings in Visualization (metabase#48497)", () => {
+    cy.window().then(win => {
+      // Monitor console errors, but don't log them in the Cypress runner UI.
+      cy.stub(win.console, "error").log(true).as("consoleError");
+    });
+
+    mountInteractiveQuestion();
+
+    cy.wait("@cardQuery").then(({ response }) => {
+      expect(response?.statusCode).to.equal(202);
+    });
+
+    // Wait until the <Visualization /> component is mounted
+    getSdkRoot().within(() => {
+      cy.findByText("Max of Quantity").should("be.visible");
+    });
+
+    cy.get<sinon.SinonStub>("@consoleError").should($console => {
+      const lifecycleErrors = $console.args.filter(args => {
+        const message = args.join(" ");
+
+        // Check that <Visualization /> is free of React lifecycle warnings
+        const visualizationHasUnsafeCycleWarning =
+          message.includes("Visualization") &&
+          message.includes("UNSAFE_component");
```

Comment:
For now we only check for the `<Visualization />` component as we fixed it in this PR, but we should add more assertions as we fix more places

Reply:
I propose adding a new mount component that automatically checks this e.g. `mountSdkContentAndAssertDeprecatedFunctions` or something. Then you could keep the whole test but change only the render function, well maybe you need to juggle around a little bit when the function is abstracted, let's say `mountInteractiveQuestion` like this, or maybe passing an option to these mount functions `mountXxx(Component, { assertDeprecatedFunctions: true })` or something. Would that be easier than manually adding assertions? I think the API would be a lot more clear that way.


Let me know what you think?

Reply:
@WiNloSt I like the idea of adding the assertion to `mountInteractiveQuestion` with `assertDeprecatedFunctions: true`! I think lifecycle errors only appears after the individual element is mounted, so we should be able to use a generic "wait for loader" command and then check if the error occurs. Let me try to revise this 🙏🏻

Reply:
fixed in https://github.com/metabase/metabase/pull/52885/commits/084167667a3648475a57fed4c8de7c61ddadde35